rclcpp/src/rclcpp/executor.cpp

```diff
@@ -39,30 +39,79 @@ using rclcpp::Executor;
 using rclcpp::ExecutorOptions;
 using rclcpp::FutureReturnCode;
 
+namespace {
+
+std::shared_ptr<rcl_guard_condition_t>
+CreateShutdownGuardCondition(std::shared_ptr<rclcpp::Context> context)
```

Comment:
nit^N: this is creating a copy of the shared pointer (inc the ref count), and that's not necessary because the function isn't retaining ownership.
Prefer `rclcpp::Context & context` (`rclcpp::Context * context` is also ok)